src/Xenial.Tasty/Remote/NamedPipeRemoteHook.cs

```diff
@@ -0,0 +1,85 @@
+using Xenial.Delicious.Scopes;
+
+namespace Xenial.Delicious.Remote
+{
+    public delegate Task<bool> IsInteractiveRun();
+
+    public delegate Task<TransportStreamFactory?> TransportStreamFactoryFunctor();
+    public delegate Task<Stream> TransportStreamFactory();
+
+    public delegate Task<TastyRemote> ConnectToRemote(TastyScope scope, Stream remoteStream);
+
+    //[EnumeratorCancellation] 
+    public delegate IAsyncEnumerable<SerializableTestCase> ReportTests(IAsyncEnumerable<SerializableTestCase> tests, CancellationToken cancellationToken);
+
+    public static class TastyRemoteDefaults
+    {
+        public static Task<bool> IsInteractiveRun()
+        {
+            var isInteractive = Environment.GetEnvironmentVariable("TASTY_INTERACTIVE");
+            if (!string.IsNullOrEmpty(isInteractive))
+            {
+                if (bool.TryParse(isInteractive, out var result))
+                {
+                    return Task.FromResult(result);
+                }
+            }
+            return Task.FromResult(false);
+        }
+
+        public static Task<TastyRemote> AttachToStream(TastyScope scope, Stream remoteStream)
+        {
+            var remote = JsonRpc.Attach<TastyRemote>(remoteStream);
+
+            scope.RegisterReporter(test => remote.Report(new SerializableTestCase
+            {
+                FullName = test.FullName,
+                Name = test.Name,
+                AdditionalMessage = test.AdditionalMessage,
+                Duration = test.Duration,
+                Exception = test.Exception,
+                IgnoredReason = test.IgnoredReason,
+                IsForced = test.IsForced?.Invoke(),
+                IsIgnored = test.IsIgnored?.Invoke(),
+                TestOutcome = test.TestOutcome
+            }));
+
+            return Task.FromResult(remote);
+        }
+    }
+
+    public static class NamedPipeRemoteHook
+    {
+        public static Task<TransportStreamFactory?> CreateNamedPipeTransportStream()
+        {
+            var connectionType = Environment.GetEnvironmentVariable("TASTY_INTERACTIVE_CON_TYPE");
+            if (connectionType != null && connectionType.ToLowerInvariant() == "NamedPipes".ToLowerInvariant())
+            {
+                var connectionId = Environment.GetEnvironmentVariable("TASTY_INTERACTIVE_CON_ID");
+                if (!string.IsNullOrEmpty(connectionId))
+                {
+                    TransportStreamFactory functor = () => CreateNamedPipeTransportStream(connectionId);
+                    return Task.FromResult<TransportStreamFactory?>(functor);
+                }
+            }
+            return Task.FromResult<TransportStreamFactory?>(null);
+        }
+
+        static async Task<Stream> CreateNamedPipeTransportStream(string connectionId)
+        {
+            var stream = new NamedPipeClientStream(".", connectionId, PipeDirection.InOut, PipeOptions.Asynchronous);
+            await stream.ConnectAsync();
```

Comment:
CancelationToken?

Reply:
I am thinking if those hooks add any value. I think it would be better to have cancellation token support baked in into the middleware, but i am a little bit afraid this will kill the transport channel.

But cause the process is controlled by CLI, i cant think of a reason why to support CTS from inside the client?